src/js/api/registries.ts

```diff
@@ -29,32 +29,46 @@ export class CommandRegistry {
 }
 
 interface Loader {
+  match: string
   load: (
     params: IngestParams & {poolId: string},
     onProgress: (value: number | null) => void,
     onWarning: (warning: string) => void,
-    onDetailUpdate: () => Promise<void>
+    onDetailUpdate: () => Promise<void>,
+    signal?: AbortSignal
   ) => Promise<void>
-  match: string
-  unLoad?: (params: IngestParams) => Promise<void>
+  unload?: (params: IngestParams) => Promise<void>
 }
 
 export class LoaderRegistry {
   private loaders: Loader[] = []
+  private emitter = new EventEmitter()
 
   constructor() {}
 
   add(l: Loader): void {
     this.loaders.push(l)
   }
-
   remove(l: Loader): void {
     if (this.loaders.includes(l)) remove(this.loaders, (l) => l === l)
   }
-
   getMatches(loadType: string): Loader[] {
     return this.loaders.filter((l) => l.match === loadType)
   }
+  onWillAbort(listener: (...args: any[]) => void): Cleanup {
```

Comment:
@jameskerr here are the api's abort-related events. I thought having explicit `willAbort` and `didAbort` methods was better than just exposing an `emit` method so that consumers don't need to know anything about internal event names ("will-abort", "did-abort", etc.). Wdyt?